substrate/frame/revive/rpc/src/lib.rs

```diff
@@ -0,0 +1,378 @@
+	}
+
+	async fn get_block_by_number(
+		&self,
+		block: BlockNumberOrTag,
+		_hydrated_transactions: bool,
+	) -> RpcResult<Option<Block>> {
+		let Some(block) = self.client.block_by_number_or_tag(&block).await? else {
+			return Ok(None);
+		};
+		let block = self.client.evm_block(block).await?;
+		Ok(Some(block))
+	}
+
+	async fn get_block_transaction_count_by_hash(
+		&self,
+		block_hash: Option<H256>,
+	) -> RpcResult<Option<U256>> {
+		let block_hash = if let Some(block_hash) = block_hash {
+			block_hash
+		} else {
+			self.client.latest_block().await.ok_or(ClientError::BlockNotFound)?.hash()
+		};
+		Ok(self.client.receipts_count_per_block(&block_hash).await.map(U256::from))
+	}
+
+	async fn get_block_transaction_count_by_number(
+		&self,
+		block: Option<BlockNumberOrTag>,
+	) -> RpcResult<Option<U256>> {
+		let Some(block) = self
+			.get_block_by_number(block.unwrap_or_else(|| BlockTag::Latest.into()), false)
+			.await?
+		else {
+			return Ok(None);
+		};
+
+		Ok(self.client.receipts_count_per_block(&block.hash).await.map(U256::from))
+	}
+
+	async fn get_storage_at(
+		&self,
+		address: H160,
+		storage_slot: U256,
+		block: BlockNumberOrTagOrHash,
+	) -> RpcResult<Bytes> {
+		let bytes = self.client.get_contract_storage(address, storage_slot, block).await?;
+		Ok(bytes.into())
+	}
+
+	async fn get_transaction_by_block_hash_and_index(
+		&self,
+		block_hash: H256,
+		transaction_index: U256,
+	) -> RpcResult<Option<TransactionInfo>> {
+		let Some(receipt) =
+			self.client.receipt_by_hash_and_index(&block_hash, &transaction_index).await
+		else {
+			return Ok(None);
+		};
+
+		let Some(signed_tx) = self.client.signed_tx_by_hash(&receipt.transaction_hash).await else {
+			return Ok(None);
+		};
+
+		Ok(Some(TransactionInfo::new(receipt, signed_tx)))
+	}
+
+	async fn get_transaction_by_block_number_and_index(
+		&self,
+		block: BlockNumberOrTag,
+		transaction_index: U256,
+	) -> RpcResult<Option<TransactionInfo>> {
+		let Some(block) = self.client.block_by_number_or_tag(&block).await? else {
+			return Ok(None);
+		};
+		self.get_transaction_by_block_hash_and_index(block.hash(), transaction_index)
+			.await
+	}
+
+	async fn get_transaction_by_hash(
+		&self,
+		transaction_hash: H256,
+	) -> RpcResult<Option<TransactionInfo>> {
+		let receipt = self.client.receipt(&transaction_hash).await;
+		let signed_tx = self.client.signed_tx_by_hash(&transaction_hash).await;
+		if let (Some(receipt), Some(signed_tx)) = (receipt, signed_tx) {
+			return Ok(Some(TransactionInfo::new(receipt, signed_tx)));
+		}
+
+		Ok(None)
+	}
+
+	async fn get_transaction_count(
+		&self,
+		address: H160,
+		block: BlockNumberOrTagOrHash,
+	) -> RpcResult<U256> {
+		let nonce = self.client.nonce(address, block).await?;
+		Ok(nonce)
+	}
+}
+
+/// A [`MiscRpcServer`] RPC server implementation.
+pub struct MiscRpcServerImpl;
+
+#[async_trait]
+impl MiscRpcServer for MiscRpcServerImpl {
+	async fn healthcheck(&self) -> RpcResult<()> {
+		Ok(())
```

Comment:
Is the idea here just to be able to ping the server?

Wondering why this is a trait, seems like if this is just to have a ping like check it seems superfluous to have a trait for it.

Reply:
I can probably remove this one now, you somehow have to define your rpc on a trait along with the `#[rpc]` macro and then implement it on the server, the client trait is generated as well for you through this Macro